src/nvim/drawline.c

```diff
@@ -1349,7 +1464,7 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
           && wp == curwin
           && lnum == wp->w_cursor.lnum
           && vcol >= (long)wp->w_virtcol)
-         || (number_only && draw_state > WL_NR))
+         || (number_only && draw_state > WL_STC))
```

Comment:
I think all these comparisons to draw state should be:

```suggestion
         || (number_only && draw_state >= WL_LINE))
```

Reply:
I don't think so. I think comparing with `WL_STC` is better.

Reply:
Good points, reverted.